pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/SubscriptionStatsImpl.java

```diff
@@ -126,15 +125,9 @@ public class SubscriptionStatsImpl implements SubscriptionStats {
     /** Whether the Key_Shared subscription mode is AUTO_SPLIT or STICKY. */
     public String keySharedMode;
 
-    /** This is for Key_Shared subscription to get the recentJoinedConsumers in the Key_Shared subscription. */
-    public Map<String, String> consumersAfterMarkDeletePosition;
```

Comment:
We should probably leave this in place, since we're going to leave the options for keeping the current behavior.

Reply:
This particular change is removing the PIP-282 implementation. In [PIP-379 backwards compatibility](https://github.com/apache/pulsar/blob/master/pip/pip-379.md#backward--forward-compatibility), we have defined the requirement this way:

> Slightly unrelated to PIP-379 changes, there's a need to ensure that users upgrading from Pulsar 3.x can revert to the "recently joined consumers" logic (before PIP-282) in case of possible regressions caused by PIP-379. Since PIP-282 is also new in Pulsar 4.0.0, there needs to be a feature flag that toggles between the PIP-379 implementation for Key_Shared and the "recently joined consumers" logic before PIP-282. Implemention details for this feature toggle can be handled in the pull request for implementing this.

The way I've been thinking of handling this is to take the PersistentStickyKeyDispatcherMultipleConsumers class from branch-3.3 and copy it with a different name in to the master branch and adapt it to the code base so that goal is achieved. The reason for this is that we want to restore the behavior of branch-3.3 with the feature flag. Master branch is using PIP-282.

Another reason is that in master branch, there are multiple other changes compared to branch-3.3 in PersistentStickyKeyDispatcherMultipleConsumers. Tracking the changes will be easier by copying the class from branch-3.3 and adapting it to match branch-3.3 behavior as much as possible.

This will require copying both PersistentDispatcherMultipleConsumers and PersistentStickyKeyDispatcherMultipleConsumers classes since both classes have changed compared to branch-3.3. 

The change set in this PR would become very messy and even larger than now if I add the changes to this same PR. That's why I think that it's better to handle in a separate PR.

Does this sound reasonable?

Reply:
As discussed with @merlimat, the stats will be restored with the PR to restore branch-3.3 behavior behind a feature flag.

Reply:
Addressed in #23424